include/xgboost/linear_updater.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018 by Contributors
+ */
+#pragma once
+
+#include <dmlc/registry.h>
+#include <xgboost/base.h>
+#include <xgboost/data.h>
+#include <functional>
+#include <string>
+#include <utility>
+#include <vector>
+#include "../../src/gbm/gblinear_model.h"
+
+namespace xgboost {
+/*!
+ * \brief interface of linear updater
+ */
+class LinearUpdater {
+ public:
+  /*! \brief virtual destructor */
+  virtual ~LinearUpdater() {}
+  /*!
+   * \brief Initialize the updater with given arguments.
+   * \param args arguments to the objective function.
+   */
+  virtual void Init(
+      const std::vector<std::pair<std::string, std::string> >& args) = 0;
+
+  /**
+   * \fn  virtual void LinearUpdater::Update(std::vector<bst_gpair>* in_gpair, DMatrix* data, gbm::GBLinearModel* model, double sum_instance_weight) = 0;
+   *
+   * \brief Updates linear model given gradients.
+   *
+   * \param in_gpair            The gradient pair statistics of the data.
+   * \param data                Input data matrix.
+   * \param model               Model to be updated.
+   * \param sum_instance_weight The sum instance weight.
+   */
+
+  virtual void Update(std::vector<bst_gpair>* in_gpair, DMatrix* data,
```

Comment:
We had discussed re-factoring `Update` interface to use `HostDeviceVector` consistently. Would this code work okay after the re-factor? My guess is that a few additional lines would be all that would be needed, but I'd like a confirmation from you.

Reply:
Or we could keep `HostDeviceVector` for tree updaters only, should we decide against future GPU support for linear boosting. Let me know what you think.

Reply:
I think it will trivially work by changing to HostDeviceVector. I do have plans for a gpu version in the near future.

Reply:
That's good to know.